ofrak_core/ofrak/core/free_space.py

```diff
@@ -131,19 +135,36 @@ async def allocate_bom(
         # Allocate largest segments first
         segments_to_allocate.sort(key=lambda o_s: o_s[1].length, reverse=True)
         segments_by_object: Dict[str, List[Segment]] = defaultdict(list)
+        possible_perms = None
         for obj, segment in segments_to_allocate:
             vaddr, final_size = 0, 0
             if segment.length > 0:
-                allocs = await self.allocate(
-                    segment.access_perms,
-                    segment.length,
-                    min_fragment_size=segment.length,
-                    alignment=bom.segment_alignment,
-                )
-                allocation = next(iter(allocs))
-                vaddr = allocation.start
-                final_size = allocation.length()
-
+                if permission_map is not None:
+                    possible_perms = permission_map[segment.access_perms]  # type: ignore
```

Comment:
```suggestion
                    possible_perms = permission_map[segment.access_perms]
```

Removing the type ignore does not seem to impact the results of running `make inspect`.